src/lib.rs

```diff
@@ -199,7 +197,7 @@ impl Obj {
 
                     // Create new object
                     let name = terms
-                        .map_while(|t| Some(t).filter(|t| util::name_is_valid(t)))
+                        .filter(|t| util::name_is_valid(t))
```

Comment:
These two pieces of code don't work the same. The original stops at an invalid name (like a comment) and the new one doesn't.

Reply:
Fixed